pennylane_lightning/src/gates/cpu_kernels/GateImplementationsLM.hpp

```diff
@@ -243,30 +259,20 @@ class GateImplementationsLM : public PauliGenerator<GateImplementationsLM> {
         }
     }
 
-  public:
     template <class PrecisionT>
-    static void applyMatrix(std::complex<PrecisionT> *arr, size_t num_qubits,
-                            const std::complex<PrecisionT> *matrix,
-                            const std::vector<size_t> &wires, bool inverse) {
-        using Util::Trans;
+    static void
+    applyMultiQubitOp(std::complex<PrecisionT> *arr, size_t num_qubits,
+                      const std::complex<PrecisionT> *matrix,
+                      const std::vector<size_t> &wires, bool inverse) {
         assert(num_qubits >= wires.size());
 
-        switch (wires.size()) {
-        case 1:
-            applySingleQubitOp(arr, num_qubits, matrix, wires[0], inverse);
-            break;
-        case 2:
-            applyTwoQubitOp(arr, num_qubits, matrix, wires, inverse);
-            break;
-        default: {
-            size_t dim = size_t{1U} << wires.size();
-            std::vector<size_t> indices;
-            indices.resize(dim);
+        size_t dim = static_cast<size_t>(1U) << wires.size();
+        std::vector<size_t> indices;
+        indices.resize(dim);
```

Comment:
Should the `dim` go into the vector constructor instead?

Reply:
Just wondering why? Providing it to the constructor directly is a safer option, no? Less lines for the same effect.

Reply:
Aha, I was thinking about another line. Yes, you are right here :sweat_smile: